cmake_test/add_section.cmake

```diff
@@ -0,0 +1,41 @@
+include_guard()
+
+
+#[[[
+# Adds a test section, should be called inside of a declared test function directly before declaring the section function.
+# The NAME parameter will be populated as by set() with the generated section function name. Declare the section function using this generated name. Ex:
+#
+# .. code-block:: cmake
+#
+#    #This is inside of a declared test function
+#    ct_add_section(NAME this_section)
+#    function(${this_section})
+#        message(STATUS "This code will run in a test section")
+#    endfunction()
+#
+# :param EXPECTFAIL: Option indicating whether the section is expected to fail or not, if specified will cause the section to be ran in a subprocess.
```

Comment:
I wasn't sure how best to document an option since it doesn't correspond with positional params or global variables, how would you like me to document such options?

Reply:
I'd try to stick with something like how kwargs are documented in Python:

```.py
:param **kwargs: See below

:Keyword Arguments:
   * *kwarg1* (``type``) -- description
   * *kwarg2* (``type``) -- description
```

This produces something which looks like how Sphinx handles the other two options. I'm open to other options, but I think it's important that they get called out differently since the user has to specify them differently (i.e as key-value pairs in an arbitrary order).

Reply:
I've reformatted the docstring for ``add_section()``, does that look right or should I adjust it some more?

Reply:
That looks good.